keras/src/quantizers/gptq_core.py

```diff
@@ -0,0 +1,335 @@
+import random
+
+import numpy as np
+from absl import logging
+
+from keras.src import ops
+from keras.src import utils as keras_utils
+from keras.src.layers import Dense
+from keras.src.layers import EinsumDense
+from keras.src.layers import Embedding
+from keras.src.quantizers.gptq import GPTQ
+from keras.src.quantizers.gptq_quant import GPTQQuantization
+
+
+def get_dataloader(tokenizer, sequence_length, dataset, num_samples=128):
+    """
+    Prepares and chunks the calibration dataloader, repeating short datasets.
+    """
+    all_tokens = []
+
+    if isinstance(dataset, str):
```

Comment:
Based on this error message, can you update the `if` to actually check that it's one of the expected type (as opposed to testing it's not one of the non-expected types)?

Reply:
Done